libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -975,6 +967,50 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 
 		break;
 	}
+	case FunctionType::Kind::ECRecover:
+	case FunctionType::Kind::RIPEMD160:
+	case FunctionType::Kind::SHA256:
```

Comment:
Maybe it makes sense to extract the whole thing into a function and call it with different arguments from the cases here?